kiwi/bootloader/config/grub2.py

```diff
@@ -1123,3 +1052,101 @@ def _get_shim_install(self):
+            #
+            # Please note this is a one time modification done by kiwi
+            # Any subsequent call of the grub config tool will overwrite
+            # the setup and disables dynamic EFI environment checking
+            # at boot time
+            with open(config_file) as grub_config_file:
+                grub_config = grub_config_file.read()
+                grub_config = re.sub(
+                    r'([ \t]+)linux(efi|16)*([ \t]+)', r'\1$linux\3',
+                    grub_config
+                )
+                grub_config = re.sub(
+                    r'([ \t]+)initrd(efi|16)*([ \t]+)', r'\1$initrd\3',
+                    grub_config
+                )
+            with open(config_file, 'w') as grub_config_file:
+                grub_config_file.write(
+                    Template(self.grub2.header_hybrid).substitute()
+                )
+                grub_config_file.write(grub_config)
+
+    def _fix_grub_root_device_reference(self, config_file, boot_options):
+        if self.root_reference:
+            if self.root_filesystem_is_overlay or \
+               Defaults.is_buildservice_worker():
+                # grub2-mkconfig has no idea how the correct root= setup is
+                # for disk images created with overlayroot enabled or in a
+                # buildservice worker environment. Because of that the mkconfig
+                # tool just finds the raw partition loop device and includes it
+                # which is wrong. In this particular case we have to patch the
+                # written config file and replace the wrong root= reference with
+                # the correct value.
+                with open(config_file) as grub_config_file:
+                    grub_config = grub_config_file.read()
+                    grub_config = grub_config.replace(
+                        'root={0}'.format(boot_options.get('root_device')),
+                        self.root_reference
+                    )
+                with open(config_file, 'w') as grub_config_file:
+                    grub_config_file.write(grub_config)
+
+                if self.firmware.efi_mode():
+                    vendor_grubenv_file = \
+                        Defaults.get_vendor_grubenv(self.efi_mount.mountpoint)
+                    if vendor_grubenv_file:
+                        with open(vendor_grubenv_file) as vendor_grubenv:
+                            grubenv = vendor_grubenv.read()
+                            grubenv = grubenv.replace(
+                                'root={0}'.format(
+                                    boot_options.get('root_device')
+                                ),
+                                self.root_reference
+                            )
+                        with open(vendor_grubenv_file, 'w') as vendor_grubenv:
+                            vendor_grubenv.write(grubenv)
+
+    def _fix_grub_loader_entries_boot_cmdline(self):
+        if self.cmdline:
```

Comment:
This execution criteria is different than it was before, I guess this is intended. So we are now applying the patch to any existing `/boot/loader/entries/*.conf` file. I am fine with this change, in fact, seams to be simpler and easier to understand, I even would say that some cases not properly covered before.... :+1: for that change

Reply:
yes I did that on purpose as I thought this is what we want as of now. The loader/entries files were so broken under several different conditions that I remember we agreed on just replacing the entire cmdline with the values we know that are correct. So the only criteria is if there is a cmdline bump it over all loader/entries

I'm happy when we can delete this though, as I think it should really work in grub independent of the host

Reply:
Thanks for the review. I'll merge and do a Staging commit. Would be great if we can check a fedora and ubuntu build if they work as before. Those are affected by the adaptions. On suse we only have the root dev update and the linuxefi vs. linux adaptions. I'll do some tests too.

Thanks